### ISSUE TITLE
Training with Adam gets slower each epoch

### ISSUE THREAD

**Issue by chusb40** [fchollet/keras#3576]
I am training a simple neural network in Keras with Theano backend consisting of 4 dense layers connected to a Merge layer and then to a softmax classifier layer. Using Adam for training, the first few epochs train in about 60s each (in the CPU) but, after that, the training time per epoch starts increasing, taking more than 400s by epoch 70, making it unusable.

Is there anything wrong with my code? Is this supposed to happen or is it a bug?

This only happens when using Adam, not with sgd, adadelta, rmsprop or adagrad.  I'd use any of the other methods but Adam produces far better results.

The code:

```
modela = Sequential()
modela.add(Dense(700, input_dim=40, init='uniform', activation='relu'))
modelb = Sequential()
modelb.add(Dense(700, input_dim=40, init='uniform', activation='relu'))
modelc = Sequential()
modelc.add(Dense(700, input_dim=40, init='uniform', activation='relu'))
modeld = Sequential()
modeld.add(Dense(700, input_dim=40, init='uniform', activation='relu'))

model = Sequential()
model.add(Merge([modela, modelb, modelc, modeld], mode='concat', concat_axis=1))
model.add(Dense(258, init='uniform', activation='softmax'))

model.compile(loss='categorical_crossentropy', optimizer='adam', metrics=['accuracy'])

hist = model.fit([Xa, Xb, Xc, Xd], Ycat, validation_split=.25, nb_epoch=80, batch_size=100, verbose=2)
```

**Comment by chusb40** [comment]
If it is of any use, I've found this problem to happen when using Adamax or Nadam too.

**Comment by keunwoochoi** [comment]
In my case it slows down a bit, but not as much as your case, FYI.

**Comment by akshaybalwally** [comment]
I'm not sure why this happens, but the issue is with the default parameters in Adam. Specifically, the beta_1 and beta_2 need to be played with. I've found that by increasing beta_1 (I went from .9 to .99), the problem can be fixed. I would ~guess~ that the slowdown occurs because the parameters or gradients or whatever decay too quickly with poorly set beta values, and you run into underflow or overflow issues, but I can't pretend to know exactly how Adam works.
tl;dr: Raise beta_1 from .9 to .99

**Comment by TruongKhang** [comment]
Same problem like you. Did you fix this?

**Comment by sachinruk** [comment]
Similar issue with Rmsprop. Any thoughts?

**Comment by adenzila** [comment]
I'm experiencing the same problem with Adam optimiser, would be interested in finding out why.

**Comment by akshaybalwally** [comment]
Maybe this would help? (not sure if you already saw this, if so sorry for reposting)
"""I'm not sure why this happens, but the issue is with the default parameters in Adam. Specifically, the beta_1 and beta_2 need to be played with. I've found that by increasing beta_1 (I went from .9 to .99), the problem can be fixed. I would guess that the slowdown occurs because the parameters or gradients or whatever decay too quickly with poorly set beta values, and you run into underflow or overflow issues, but I can't pretend to know exactly how Adam works.
tl;dr: Raise beta_1 from .9 to .99"""

**Comment by jsanket9578** [comment]
@chusb40 Did you get a solution to this problem?

**Comment by Owen864720655** [comment]
Do anybody have a solution?

**Comment by matankley** [comment]
Any solution ? Same happens for me with Adam makes the model unstable for use
